### ISSUE TITLE
add profile picture

### ISSUE THREAD

**Issue by 4kh00** [jerimiah-smiggins/smiggins#268]
Okay, so I think there should be a way to set profile pictures. Of course, there should be an admin vetting option to stop sickos from posting sicko things. The banner thing does look cool, though.

**Comment by trinkey** [comment]
i won't be doing this (at least for now) for the following reasons:
- image uploading is a huge pain. you need to have a file input but also like unless you wanna do hellish javascript stuff it'd need a form submit, and if you do that it gets asynchronously uploaded to the server meaning if you have a slow connection and/or large file itll just be loading for a long time. also nginx would need to be configured to allow larger payloads if that's setup
- you'd need a place to store the images. on pythonanywhere for free accounts, you only get 512mb of storage, ~120mb gets taken up by django. not even mentioning how you'll likely want to setup a cdn to deliver images for you, like an aws s3 bucket, all of which would need to be implemented and compatability tested.
- image compression on the server is expensive. assuming you don't want images to be like 1000tb each, youll want to compress them, which takes a decent amount of processing power. this isn't a huge issue if you have multiple cores dedicated to the server, so it won't affect other users, however free pythonanywhere accounts and development servers only have one worker dedicated to the server.
- image validation is difficult. what's stopping you from uploading a garbage image to the server that doesn't even work? validating images is usually done only by "magic numbers" (special bytes at the beginning of files), however those can easily be faked. in order to make sure images aren't broken/malformed, there would likely need to be a check for every file type that is supported, ex. png, jpg, webp, gif, etc.
- if the user wants to crop their pfp, that would need to be done in the frontend, and making responsive canvas stuff for cropping images or like anything using html canvas is a massive pain
- less of an issue, but if all pfps went through an admin portal for verification, it could cause delays for the end user. what if they uploaded the wrong image and they need to change it?

tldr: pain and suffering.

feel free to implement this and make a pr though if you want

**Comment by trinkey** [comment]
> It looks like magic.

completely understood
